indexer/services/roundtable/src/tasks/update-affiliate-info.ts

```diff
@@ -0,0 +1,50 @@
+import { logger, stats } from '@dydxprotocol-indexer/base';
+import {
+  PersistentCacheTable, AffiliateInfoTable, PersistentCacheKeys, PersistentCacheFromDatabase,
+} from '@dydxprotocol-indexer/postgres';
+import { DateTime } from 'luxon';
+
+import config from '../config';
+
+const defaultLastUpdateTime: string = '2023-10-26T00:00:00Z';
+
+/**
+ * Update the affiliate info for all affiliate addresses.
+ */
+export default async function runTask(): Promise<void> {
+  try {
+    const start = Date.now();
+    const persistentCacheEntry: PersistentCacheFromDatabase | undefined = await PersistentCacheTable
+      .findById(PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME);
+
+    if (!persistentCacheEntry) {
+      logger.info({
+        at: 'update-affiliate-info#runTask',
+        message: `No previous ${PersistentCacheKeys.AFFILIATE_INFO_UPDATE_TIME} found in persistent cache table. Will use default value: ${defaultLastUpdateTime}`,
+      });
+    }
+
+    const lastUpdateTime: DateTime = DateTime.fromISO(persistentCacheEntry
+      ? persistentCacheEntry.value
+      : defaultLastUpdateTime);
+    let windowEndTime = DateTime.utc();
```

Comment:
Upper bound should not be at DateTime.utc(), it should be set at the timestamp of the latest block height, because if the indexer is stalled, we can only update affiliate info up to the latest block time. See [here](https://github.com/dydxprotocol/v4-chain/blob/main/indexer/services/roundtable/src/tasks/aggregate-trading-rewards.ts#L153) as an example, you should also change update-wallet-total-volume to match the logic

Reply:
Fixed